plugin.rb

```diff
@@ -95,15 +100,12 @@
       end
   end
 
-  add_to_serializer(:post, :include_preloaded_surveys?) { false }
-
-  add_to_serializer(:post, :surveys, respect_plugin_enabled: false) do
-    preloaded_surveys.map { |s| SurveySerializer.new(s, root: false, scope: self.scope) }
-  end
-
-  add_to_serializer(:post, :include_surveys?) do
-    SiteSetting.surveys_enabled && preloaded_surveys.present?
-  end
+  add_to_serializer(
+    :post,
+    :surveys,
+    respect_plugin_enabled: false,
+    include_condition: -> { SiteSetting.surveys_enabled && preloaded_surveys.present? },
```

Comment:
this should be equivalent (but shorter) right?

```suggestion
    respect_plugin_enabled: true,
    include_condition: -> { preloaded_surveys.present? },
```

Reply:
After some archaeology, it seems to be the case.